lib/pure/os.nim

```diff
@@ -296,6 +296,27 @@ proc setCurrentDir*(newDir: string) {.inline, tags: [].} =
   else:
     if chdir(newDir) != 0'i32: raiseOSError(osLastError())
 
+proc absolutePath*(path: string, root = getCurrentDir()): string =
+  ## Returns the absolute path of `path`, rooted at `root` (which must be absolute)
+  ## if `path` is absolute, return it, ignoring `root`
+  runnableExamples:
+    doAssert absolutePath("a") == getCurrentDir() / "a"
+  if isAbsolute(path): path
+  else:
+    if not root.isAbsolute:
+      # CHECKME: is that the correct exception type?
+      raise newException(ValueError, root)
```

Comment:
This exception type seems fine, but the error message is poor. You need something like "The specified `root` is not absolute: <root>"